source/student-orgs/presence.ts

```diff
@@ -0,0 +1,81 @@
+import {get, http} from '../ccc-lib/http.js'
+import lodash from 'lodash'
+import {JSDOM} from 'jsdom'
+import pMap from 'p-map'
+import {z} from 'zod'
+import {SortableStudentOrgSchema, type SortableStudentOrgType} from './types.js'
+
+const BasicPresenceOrgSchema = z.object({
+	subdomain: z.string(),
+	campusName: z.string(),
+	name: z.string(),
+	uri: z.string(),
+	regularMeetingTime: z.string(),
+	regularMeetingLocation: z.string(),
+	hasCoverImage: z.boolean(),
+	photoUri: z.string(),
+	photoUriWithVersion: z.string(),
+	photoType: z.union([z.literal('upload'), z.string()]),
+	memberCount: z.number(),
+	categories: z.string().array(),
+	newOrg: z.boolean(),
+	hasUpcomingEvents: z.boolean(),
+})
+
+type DetailedPresenceOrgType = z.infer<typeof DetailedPresenceOrgSchema>
+const DetailedPresenceOrgSchema = BasicPresenceOrgSchema.and(
+	z.object({
+		description: z.string(),
+		website: z.string().optional(),
+	}),
+)
+
+export function cleanOrg(org: DetailedPresenceOrgType, sortableRegex: RegExp) {
+	let name = org.name.trim()
+	let category = org.categories.join(', ')
+	let meetings =
+		(org.regularMeetingLocation || '').trim() +
+		(org.regularMeetingTime || '').trim()
+	let description = JSDOM.fragment(org.description).textContent?.trim() ?? ''
+	let website = org.website?.trim() ?? ''
+	if (website && !/^https?:\/\//.test(website)) {
+		website = `http://${website}`
```

Comment:
Why did we opt for http again?

Reply:
At the time, I believe there were several sites that only supported HTTP. I think we could probably do HTTPS now, without breaking anyone…